lib/faraday/response.rb

```diff
@@ -15,8 +15,12 @@ def call(env)
 
       # Override this to modify the environment after the response has finished.
       # Calls the `parse` method if defined
+      # `parse` method can be defined as private, public and protected
       def on_complete(env)
-        env.body = parse(env.body) if respond_to?(:parse) && env.parse_body?
+        all_included = true
+        return unless respond_to?(:parse, all_included) && env.parse_body?
```

Comment:
Any reason for defining `all_included` as a separate variable and then using it on the next line?
Why not just using `respond_to?(:parse, true)`? People should know what the 2nd param for `respond_to?` is.

Reply:
I added the separate variable because to me was a way to increase the readability of the code since i was not expecting everybody to know signature for `respond_to?`.

That been said i don't think it is necessary so I am happy to apply to your change request.

:-)